File: setup_claude.py
#!/usr/bin/env python3
import json
import shutil
from pathlib import Path

CLAUDE_SETTINGS = Path.home() / '.claude' / 'settings.json'
SCRIPT_PATH = str(Path.home() / 'scripts' / 'aiagent_notify.py')

def setup_hooks():
    """配置 Claude Code hooks"""
    if not CLAUDE_SETTINGS.exists():
        print(f"❌ {CLAUDE_SETTINGS} 不存在")
        return False

    # 备份
    backup = CLAUDE_SETTINGS.with_suffix('.json.bak')
    shutil.copy(CLAUDE_SETTINGS, backup)
    print(f"✅ 已备份到 {backup}")

    # 读取配置
    with CLAUDE_SETTINGS.open() as f:
        config = json.load(f)

    # 配置 hooks
    hook_config = {
        "matcher": "",
        "hooks": [{
            "type": "command",
            "command": f"python3 {SCRIPT_PATH}",
            "timeout": 5
        }]
    }

    if 'hooks' not in config:
        config['hooks'] = {}

    config['hooks']['Stop'] = [hook_config]
    config['hooks']['SessionEnd'] = [hook_config]

    # 写入配置
    with CLAUDE_SETTINGS.open('w') as f:
        json.dump(config, f, indent=2)

    print(f"✅ 已配置 hooks: {SCRIPT_PATH}")
    return True

if __name__ == '__main__':
    print("=== Claude Code 配置工具 ===\n")
    if setup_hooks():
        print("\n✅ 配置完成！")
    else:
        print("\n❌ 配置失败")

File: src/setup-claude.ts
import fs from 'fs';
import os from 'os';
import path from 'path';
import { fileURLToPath } from 'url';

const { promises: fsp } = fs;

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const claudeDir = path.join(os.homedir(), '.claude');
const claudeSettingsPath = path.join(claudeDir, 'settings.json');
const backupSettingsPath = path.join(claudeDir, 'settings.json.bak');
// 动态解析路径：支持 npm link 全局安装和本地开发
const notifyScriptPath = path.resolve(__dirname, 'aiagent-notify.js');
const hookCommand = `node ${notifyScriptPath}`;
const hookNames = ['Stop', 'SessionEnd'] as const;

interface ClaudeHook {
  matcher: string;
  hooks: Array<{
    type: 'command';
    command: string;
    timeout: number;
  }>;
}

type ClaudeConfig = {
  hooks?: Record<string, ClaudeHook[]>;
  [key: string]: unknown;
};

async function setupHooks(): Promise<boolean> {
  if (!fs.existsSync(claudeSettingsPath)) {
    console.error(`❌ ${claudeSettingsPath} 不存在`);
    return false;
  }

  try {
    await fsp.copyFile(claudeSettingsPath, backupSettingsPath);
    console.log(`✅ 已备份到 ${backupSettingsPath}`);
  } catch (error) {
    console.error('❌ 备份失败：', error);
    return false;
  }

  let config: ClaudeConfig;
  try {
    const raw = await fsp.readFile(claudeSettingsPath, 'utf-8');
    config = JSON.parse(raw);
  } catch (error) {
    console.error('❌ 读取配置失败：', error);
    return false;
  }

  const hookConfig: ClaudeHook = {
    matcher: '',
    hooks: [
      {
        type: 'command',
        command: hookCommand,
        timeout: 5,
      },
    ],
  };

  config.hooks ??= {};
  for (const name of hookNames) {
    config.hooks[name] = [hookConfig];
  }

  try {
    await fsp.writeFile(claudeSettingsPath, JSON.stringify(config, null, 2));
    console.log(`✅ 已配置 hooks: ${hookCommand}`);
    return true;
  } catch (error) {
    console.error('❌ 写入配置失败：', error);
    return false;
  }
}

async function main() {
  console.log('=== Claude Code 配置工具 ===\n');
  const success = await setupHooks();
  if (success) {
    console.log('\n✅ 配置完成！');
  } else {
    console.log('\n❌ 配置失败');
  }
}

void main();
